### DISCUSSION TITLE
"Unknown Kotlin JVM target"

### DISCUSSION THREAD

**Discussion by s-h-a-d-o-w** [tauri-apps/tauri#10315] [Q&A] [answered]
I keep getting this error when trying to use `dev` for Android but no indication whatsoever which version works. First, I tried to 22, now 21 based on this:

```
Welcome to Gradle 8.4!

Here are the highlights of this release:
 - Compiling and testing with Java 21
...
```

I don't want to keep going down version by version... can anybody please tell me which one works? At this point, I'm not sure that it's really related to the gradle version. (I'd also recommend documenting this.)

**Comment by FabianLars** [comment, marked as answer]
> I don't want to keep going down version by version... can anybody please tell me which one works?

17 works for sure.

**Reply by s-h-a-d-o-w** [reply]
Thanks!

**Reply by rcorre** [reply]
To clarify, we should install and use java17, rather than using the java shipped with android studio (in `/opt/android-studio/jbr/bin`)? I was a bit confused by this since the [instructions](https://v2.tauri.app/start/prerequisites/#android) say:

```
export JAVA_HOME=/opt/android-studio/jbr
```

**Reply by FabianLars** [reply]
the android studio one should work perfectely, many people (like OP i guess) try to use an external java installation though.

**Reply by rcorre** [reply]
Hmm, I was using the one from `android-studio` at first though, and got the same error (`/opt/android-studio/jbr/bin/java` is Java 21).

**Reply by FabianLars** [reply]
hmm, i downgraded back to android studio 2024.1 shortly after release for an unrelated issue so i didn't test 2024.2 (which updated to java21) much. 17 still has the best support (same for other ecosystems from what i've seen in google search results just now), but i thought i fixed all compatiblity issues with 21 by [upgrading stuff](https://github.com/tauri-apps/tauri/pull/10507).

Maybe we should add a note to the preqreqs docs for the time being just to be safe. I'll also try 21 myself again in the next few days.

**Reply by rcorre** [reply]
Thanks! I worked around it for now by installing java17 and pointing JAVA_HOME at that

**Reply by FabianLars** [reply]
Not really helpful but i re-installed android studio and got the latest version this time (the one that includes java 21) and it seems to work fine for me. It does require regenerating the android project though, so if you tried a project created with java 17 you may hit issues.
